apps/omg_watcher/lib/omg_watcher/web/controllers/transaction.ex

```diff
@@ -47,10 +47,27 @@ defmodule OMG.Watcher.Web.Controller.Transaction do
   Submits transaction to child chain
   """
   def submit(conn, params) do
-    with {:ok, tx} <- expect(params, "transaction", :hex) do
-      API.Transaction.submit(tx)
-      |> api_response(conn, :submission)
+    elem =
+      with {:ok, tx} <- expect(params, "transaction", :hex) do
+        API.Transaction.submit(tx)
+        |> api_response(conn, :submission)
+      end
+
+    case elem do
+      {:error, {:validation_error, _, _}} ->
+        Appsignal.increment_counter("transaction.failed.validation", 1)
+
+      {:error, {error, _}} when is_atom(error) ->
+        Appsignal.increment_counter("transaction.fail." <> Atom.to_string(error), 1)
+
+      {:error, _} ->
+        Appsignal.increment_counter("transaction.fail.unidentified", 1)
+
+      _ ->
```

Comment:
what is this supposed to handle?

Reply:
question still stands, what is `_`? are you sure you're not catching `everything else`?

Reply:
everything else is success

Reply:
I hope you're correct ;)

Reply:
my advice would be that you always prefer explicit over implicit. makes it more readable and reliable. especially since there's no dialyzer specs.

Reply:
in order to calm you down, I suggest this solution:
```elixir
  defp increment_metrics_counter(response) do
    case response do
      {:error, {:validation_error, _, _}} ->
        Appsignal.increment_counter("transaction.failed.validation", 1)

      {:error, {error, _}} when is_atom(error) ->
        Appsignal.increment_counter("transaction.fail." <> Atom.to_string(error), 1)

      %Plug.Conn{} ->
        Appsignal.increment_counter("transaction.succeed", 1)

      _ ->
        Appsignal.increment_counter("transaction.fail.unidentified", 1)
    end

    response
  end
```